src/ServicePulse.Host/vue/src/components/monitoring/EndpointListSortableColumn.vue

```diff
@@ -1,19 +1,81 @@
 <script setup>
-defineProps({
+import { ref, computed } from "vue";
+import { useMonitoringStore } from "../../stores/MonitoringStore";
+
+const props = defineProps({
   unit: String,
+  isActive: Boolean,
+  sortBy: {
+    type: String,
+    required: true,
+    validator: (value) => ["name", "queueLength", "throughput", "retries", "processingTime", "criticalTime"].includes(value),
+  },
 });
+
+const monitoringStore = useMonitoringStore();
+const isActive = computed(() => props.isActive);
+const sortBy = computed(() => props.sortBy);
+const isAscending = ref(false);
+const sortIcon = ref("sort-down");
```

Comment:
This could be a computed property off of `isAscending` and then you don't need to change it below

```suggestion
const sortIcon = computed(() => (isAscending.value ? "sort-up" : "sort-down"));
```